### PR TITLE
QF-115 Employer Feedback via MA

### PR DESCRIPTION
**Pull request by techtoniq**

### CODE REVIEW

File: src/Employer/ESFA.DAS.EmployerProvideFeedback/Views/Confirmation/Index.cshtml [hunk 1]
```diff
@@ -1,37 +1,43 @@
...
+                    <p class="govuk-body">If you have concerns about @Model.ProviderName, tell your line manager or Human Resources team - your organisation might be able to solve your problem informally.</p>
+                    <p class="govuk-body">If you cannot resolve it informally, you can make a formal complaint to your training provider.</p>
+                    <p class="govuk-body">If you've done this and you're not satisfied with the result, you can ask ESFA to investigate how the provider handled your case. ESFA can't review the result of your complaint - it can only investigate whether it was reached in a fair way.</p>
+                </div>
+            </details>
+            }
+            <h2 class="govuk-heading-m">What do you want to do next?</h2>
+            @if(Model.HasMultipleProviders)
+            {
+                <p class="govuk-body"><a href="/@Model.EncodedAccountId/providers" class="govuk-link--no-visited-state">Give feedback on another training provider</a></p>
```
Comment (jamesmking) [inline src/Employer/ESFA.DAS.EmployerProvideFeedback/Views/Confirmation/Index.cshtml:37]: These links will need the `.govuk-link` css class as well, so it reads `class="govuk-link govuk-link--no-visited-state"`

File: src/Employer/ESFA.DAS.EmployerProvideFeedback/Views/Confirmation/Index.cshtml [hunk 2]
```diff
@@ -1,37 +1,43 @@
...
+                <p class="govuk-body">or</p>
+            }
+            <p class="govuk-body"><a href="/" class="govuk-link--no-visited-state">Return home</a></p>
```
Comment (jamesmking) [inline src/Employer/ESFA.DAS.EmployerProvideFeedback/Views/Confirmation/Index.cshtml:40]: `govuk-link` class needed on this link as well

File: src/Employer/ESFA.DAS.EmployerProvideFeedback/Views/FeedbackSubmitted/Index.cshtml [hunk 3]
```diff
@@ -1,16 +1,17 @@
...
-        <p class="lede">You'll get chance to leave feedback in a few months.</p>
-        <p>You can <a href="@externalLinks.Value.ManageApprenticeshipSiteUrl">sign into your apprenticeship account</a>.</p>
+<main class="govuk-main-wrapper govuk-main-wrapper--auto-spacing">
+    <div class="govuk-grid-row">
+        <div class="govuk-grid-column-two-thirds">
+            <h1 class="govuk-heading-l">
+                Feedback already submitted
+            </h1>
+            <p class="govuk-body">You'll get chance to leave feedback in a few months.</p>
+            <p class="govuk-body">You can <a href="@externalLinks.Value.ManageApprenticeshipSiteUrl">sign into your apprenticeship account</a>.</p>
```
Comment (jamesmking) [inline src/Employer/ESFA.DAS.EmployerProvideFeedback/Views/FeedbackSubmitted/Index.cshtml:14]: govuk-link class needed on this link

File: src/Employer/ESFA.DAS.EmployerProvideFeedback/Views/Provider/ConfirmProvider.cshtml [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+                            <h3 class="govuk-fieldset__heading">
+                                Is this the training provider you want to give feedback on?
+                            </h3>
+                        </legend>
+                        @Html.ValidationMessageFor(m => m.Confirmed, $"Please select an option", new { @class = "govuk-error-message" })
+                        <div class="govuk-radios" data-module="govuk-radios">
+                            <div class="govuk-radios__item">
+                                @Html.RadioButtonFor(model => model.Confirmed, "true", new { id = "yes", @class = "govuk-radios__input" })
+
+                                <label class="govuk-label govuk-radios__label" for="where-do-you-live">
```
Comment (jamesmking) [inline src/Employer/ESFA.DAS.EmployerProvideFeedback/Views/Provider/ConfirmProvider.cshtml:59]: The `for` attribute needs to match the ID of the radio button

File: src/Employer/ESFA.DAS.EmployerProvideFeedback/Views/Provider/ConfirmProvider.cshtml [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+                                    Yes
+                                </label>
+                            </div>
+                            <div class="govuk-radios__item">
+
+                                @Html.RadioButtonFor(model => model.Confirmed, "false", new { id = "no", @class = "govuk-radios__input" })
+
+
+                                <label class="govuk-label govuk-radios__label" for="where-do-you-live-2">
```
Comment (jamesmking) [inline src/Employer/ESFA.DAS.EmployerProvideFeedback/Views/Provider/ConfirmProvider.cshtml:68]: The `for` attribute needs to match the ID of the radio button

File: src/Employer/ESFA.DAS.EmployerProvideFeedback/Views/Provider/_SelectProviderPartial.cshtml [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+                }
+            </div>
+        </div>
+    </div>
+</form>
+<div class="govuk-grid-row govuk-!-margin-bottom-2">
+    <div class="govuk-grid-column-two-thirds">
+        <p class="govuk-body">
+            @Model.TrainingProviders.TotalRecordCount training provider record@{if (Model.TrainingProviders.TotalRecordCount != 1)
+                { <text>s</text> } }
```
Comment (jamesmking) [inline src/Employer/ESFA.DAS.EmployerProvideFeedback/Views/Provider/_SelectProviderPartial.cshtml:53]: `<text>` isn't valid html, use `<span>` instead

File: src/Employer/ESFA.DAS.EmployerProvideFeedback/Views/Questions/QuestionOne.cshtml [hunk 7]
```diff
@@ -3,48 +3,47 @@
...
-        {
-            <div class="error-summary" role="alert" aria-labelledby="error-summary-heading" tabindex="-1">
-                @*<h2 class="heading-medium error-summary-heading" id="error-summary-heading">
+<a asp-controller="Home" asp-action="Index" class="govuk-back-link">Back</a>
+<main class="govuk-main-wrapper govuk-main-wrapper--auto-spacing">
+    <div class="govuk-grid-row">
+        <div class="govuk-grid-column-two-thirds">
+            @if (!ViewData.ModelState.IsValid)
+            {
+                <div class="error-summary" role="alert" aria-labelledby="error-summary-heading" tabindex="-1">
```
Comment (jamesmking) [inline src/Employer/ESFA.DAS.EmployerProvideFeedback/Views/Questions/QuestionOne.cshtml:12]: These CSS classes are not correct

File: src/Employer/ESFA.DAS.EmployerProvideFeedback/Views/Questions/QuestionThree.cshtml [hunk 8]
```diff
@@ -1,56 +1,59 @@
...
-            <div class="form-group">
-                <fieldset class="@(!ViewData.ModelState.IsValid ? "form-group-error" : "")">
-                    <!-- dynamically add this class in fieldset in case of error-->
-                    <legend>
-                        <h2 class="heading-medium">How would you rate @(Model.ProviderName.EndsWith("'s") ? @Model.ProviderName : $"{@Model.ProviderName}'s") training and service overall?</h2>
-                    </legend>
+            <div class="govuk-form-group error-message-container" sfa-anyvalidationerror-class="govuk-form-group--error">
+                <h1 class="govuk-heading-l">Overall rating</h1>
+                <h2 class="govuk-heading-m">How would you rate @(Model.ProviderName.EndsWith("'s") ? @Model.ProviderName : $"{@Model.ProviderName}'s") training and service overall?</h2>
+                <fieldset name="doing-well" aria-describedby="doing-well-hint" class="govuk-fieldset checkbox-limit @(!ViewData.ModelState.IsValid ? "form-group-error" : "")">
```
Comment (jamesmking) [inline src/Employer/ESFA.DAS.EmployerProvideFeedback/Views/Questions/QuestionThree.cshtml:28]: A `<fieldset>` should always a `<legend>` - i suggest moving the `<h2>` from above

File: src/Employer/ESFA.DAS.EmployerProvideFeedback/Views/ReviewAnswers/Index.cshtml [hunk 9]
```diff
@@ -1,80 +1,91 @@
...
+<main class="govuk-main-wrapper govuk-main-wrapper--auto-spacing">
+    <div class="govuk-grid-row">
+        <div class="govuk-grid-column-two-thirds">
+
+            <form asp-route="@RouteNames.ReviewAnswers_Post">
+                <div class="govuk-form-group">
+                    <h1 class="govuk-heading-l">Check your answers</h1>
+                    <h2 class="govuk-heading-m">Training provider: @Model.ProviderName</h2>
+
+                    <fieldset name="check-your-answers" class="govuk-fieldset checkbox-limit @(!ViewData.ModelState.IsValid ? "form-group-error" : "")">
```
Comment (jamesmking) [inline src/Employer/ESFA.DAS.EmployerProvideFeedback/Views/ReviewAnswers/Index.cshtml:14]: Fieldset needs a legend

File: src/Employer/ESFA.DAS.EmployerProvideFeedback/Views/Provider/_SelectProviderPartial.cshtml (hunk 6 above)
Comment (techtoniq) [reply src/Employer/ESFA.DAS.EmployerProvideFeedback/Views/Provider/_SelectProviderPartial.cshtml:53]: <text> is valid when inside a C# code block in order to output html

Comment (jamesmking) [reply src/Employer/ESFA.DAS.EmployerProvideFeedback/Views/Provider/_SelectProviderPartial.cshtml:53]: Ah, fair enough, I didn't realise they were not rendered

File: src/Employer/ESFA.DAS.EmployerProvideFeedback/Views/Questions/QuestionOne.cshtml (hunk 7 above)
Comment (techtoniq) [reply src/Employer/ESFA.DAS.EmployerProvideFeedback/Views/Questions/QuestionOne.cshtml:12]: Could you clarify which CSS classes are not correct and what the correct ones should be please?

Comment (jamesmking) [reply src/Employer/ESFA.DAS.EmployerProvideFeedback/Views/Questions/QuestionOne.cshtml:12]: Sorry, should have said. The classes on the error summary  and the child elements are all missing the `govuk-` prefix

**Review by CodeWolf86 — APPROVED** [review]

**Pull request merged**